src/jukebox/components/synchronisation/sync_shared/__init__.py

```diff
@@ -0,0 +1,368 @@
+            _cleaned_foldername = self._clean_foldername(_music_library_path, folder)
+            _src_path = self._ensure_trailing_slash(os.path.join(_sync_remote_path_audio, _cleaned_foldername))
+            # TODO fix general absolut/relativ folder path handling
+            _dst_path = self._ensure_trailing_slash(os.path.join(_music_library_path, folder))
+
+            if self._isdir_remote(_src_path):
+                _files_synced = self._sync_paths(_src_path, _dst_path)
+
+                if _files_synced:
+                    logger.debug('Files synced: update database')
+                    plugs.call_ignore_errors('player', 'ctrl', 'update_wait')
+
+            else:
+                logger.warn(f"Folder does not exist remote: {_src_path}")
+
+        return _files_synced
+
+    def _sync_paths(self, src_path: str, dst_path: str) -> bool:
+        _files_synced = False
+        logger.debug(f"Src: '{src_path}' -> Dst: '{dst_path}'")
+
+        if dst_path.endswith('/'):
+            os.makedirs(dst_path, exist_ok=True)
+
+        if self._sync_is_mode_ssh:
+            _user = self._sync_remote_ssh_user
+            _host = self._sync_remote_server
+            _port = self._sync_remote_port
+
+            _paths = ['-e', f"ssh -p {_port}", f"{_user}@{_host}:'{src_path}'", dst_path]
+
+        else:
+            _paths = [src_path, dst_path]
+
+        _run_params = (['rsync',
+                        '--compress', '--recursive', '--itemize-changes',
+                        '--safe-links', '--times', '--omit-dir-times',
+                        '--delete', '--prune-empty-dirs',
+                        '--exclude=folder.conf',  # exclude if existing from v2.x
+                        '--exclude=.*', '--exclude=.*/', '--exclude=@*/', '--cvs-exclude'
+                        ] + _paths)
+
+        _runresult = subprocess.run(_run_params, shell=False, check=False, capture_output=True, text=True)
+
+        if _runresult.returncode == 0 and _runresult.stdout != '':
+            logger.debug(f"Synced:\n{_runresult.stdout}")
+            _files_synced = True
+        if _runresult.stderr != '':
+            logger.error(f"Sync Error: {_runresult.stderr}")
+
+        return _files_synced
+
+    def _is_server_reachable(self) -> bool:
+        _host = self._sync_remote_server
+        _port = self._sync_remote_port
+        _timeout = self._sync_remote_timeout
+
+        try:
+            sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            sock.settimeout(_timeout)
+            result = sock.connect_ex((_host, _port))
+        except Exception as e:
+            logger.error(f"Server not reachable: {_host}:{_port}. {e.__class__.__name__}: {e}")
+            return False
+
+        _server_reachable = result == 0
+        if not _server_reachable:
+            logger.error(f"Server not reachable: {_host}:{_port}. errorcode: {result}")
+
+        return _server_reachable
+
+    def _isfile_remote(self, path: str) -> bool:
```

Comment:
`_is_file_remote` .. I would separate `is` and `file`

Reply:
done